kafka/src/main/java/com/metamx/tranquility/kafka/Main.java

```
@@ -0,0 +1,149 @@
+    try {
+      main = SingleCommand.singleCommand(Main.class).parse(args);
+    }
+    catch (Exception e) {
+      log.error(e, "Exception parsing arguments");
+      Help.help(SingleCommand.singleCommand(Main.class).getCommandMetadata());
+      return;
+    }
+
+    if (main.helpOption.showHelpIfRequested()) {
+      return;
+    }
+
+    main.run();
+  }
+
+  public void run() throws InterruptedException
+  {
+    if (propertiesFile == null || propertiesFile.isEmpty()) {
+      helpOption.help = true;
+      helpOption.showHelpIfRequested();
+
+      log.warn("Missing required parameters, aborting.");
+      return;
+    }
+
+    Tuple3<TranquilityKafkaConfig, scala.collection.immutable.Map<String, DataSourceConfig<TranquilityKafkaConfig>>,
+        scala.collection.immutable.Map<String, Object>> configs = null;
+    try (InputStream is = new FileInputStream(propertiesFile)) {
+      configs = ConfigHelper.readConfigYaml(is, TranquilityKafkaConfig.class);
+    }
+    catch (IOException e) {
+      log.error("Could not read config file: %s, aborting.", propertiesFile);
+      Throwables.propagate(e);
+    }
+
+    TranquilityKafkaConfig globalConfig = configs._1();
+    Map<String, DataSourceConfig<TranquilityKafkaConfig>> datasourceConfigs = JavaConversions.mapAsJavaMap(configs._2());
+    Map<String, Object> globalProperties = JavaConversions.mapAsJavaMap(configs._3());
+
+    // find all properties that start with 'kafka.' and pass them on to Kafka
+    final Properties kafkaProperties = new Properties();
+    for (Map.Entry<String, Object> entry : globalProperties.entrySet()) {
+      String key = entry.getKey();
+      log.info("%s = %s", key, globalProperties.get(key).toString());
+      if (key.startsWith("kafka.")) {
+        kafkaProperties.setProperty(key.replaceFirst("kafka.", ""), globalProperties.get(key).toString());
+      }
+    }
+
+    // set the critical Kafka configs again from TranquilityKafkaConfig so it picks up the defaults
```

This should probably throw an error if a property is specified in both places but have different values, since otherwise people with mismatched configs may have a hard time debugging.